### PR TITLE
Addition of new notebook for BQML LLM examples

### PR DESCRIPTION
**Pull request by rachael-ds**
Adding new notebook to showcase how to use LLMs in BQML.
Examples are meant to be short, snappy and run in isolation to showcase at a high level what is possible in BQML.
Linter tests passed successfully.

### CODE REVIEW

File: language/examples/bqml/bqml_llm_examples.ipynb [hunk 1]
```diff
@@ -0,0 +1,944 @@
+{
```
Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>It would be helpful to explain "why BigQuery ML here" to elaborate on the unique value that using LLMs within BigQuery can provide to the readers/users of this notebook.</p><p> </p><p>For example, "With many organizations storing their data in BigQuery, you can now use Vertex AI LLMs directly in BigQuery, without needing to move your data."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>After pip installing, there should also be a cell to show how to restart the kernel/runtime, to allow the library to reload and be recognized. Could you copy/paste the relevant restart kernel cells from here: https://github.com/GoogleCloudPlatform/generative-ai/blob/main/language/intro_palm_api.ipynb</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Could you add in a blurb here on what the user should do explicitly in the cell below and what they mean?</p><p> </p><p>For example:</p><p> </p><p>Fill in the following variable names below:</p><ul><li>**DATASET_ID**: an ID of an existing BigQuery dataset</li><li>**CONN_NAME**: the name of a BigQuery connector that you will create later in this notebook</li><li>**CONN_SERVICE_ACCOUNT**: the email address of the service account that you will use for your BigQuery connection</li><li>**LLM_MODEL_NAME**:__&lt;description&gt;___</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>As some user may use Vertex AI Workbench, they would not need to authenticate their notebook environment. Could you copy the section on authentication from this notebook? https://github.com/GoogleCloudPlatform/generative-ai/blob/main/language/intro_palm_api.ipynb</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Could you test this in Vertex AI Workbench as well? I believe this may not work directly in Workbench (b/c it uses a service account which may not have sufficient permission to add IAM policy bindings. So the code may need to be run outside of Workbench, such as in Cloud Shell. If so, it would be important to add in some instructions on how to run these gcloud commands in Cloud Shell).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Headers will need a space after the ## symbols for it to render.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>The word "create" in "create the LLM model" might be confusing as it sounds like the user is creating/training their own LLM model, which they aren't -- instead they're creating a BigQuery ML model that is a remote connection to the Vertex AI LLM. Perhaps it would be better to rephrase this to be clearer.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after ### </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after ### </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>This cell doesn't produce an output in the notebook, but I'd certainly recommend showing the result directly here, as in adding <code>result.to_dataframe()</code> after <code>result = run_bq_query(sql)</code> (and thus remove the <code>result.to_dataframe()</code> code below).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>It might be good to talk a little bit or reiterate what just happened -- that the LLM was used to describe a cat in one paragraph.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after ##.</p><p> </p><p>Also I'd just simplify the header title to: <code>## Example use cases</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after #####.</p><p> </p><p>I think it'd be great to see a brief blurb here (and for each of the examples later on in this notebook) before jumping directly into the code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Because this is a complex query, I'd actually suggest breaking this up into two or more cells:</p><p> </p><ol><li>The raw data, before passing it to the LLM.</li><li>The prompt template</li><li>Combining the raw data + prompt template + LLM call, using something like an f-string or &lt;string&gt;.format syntax</li></ol><p> </p><p>By doing so, you can differentiate between the before/after of the LLM call, and you also show how a user can carefully craft the prompt as a separate step before using it with the LLM </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after #### </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Again here it might be better to show the <em>before</em> vs the <em>after. You can even do a count of the character or word length before/after to show the % reduction via summarization and the final word count.</em></p><p> </p><p><em>The LLM may not always return a summary within 25 words or less, so it may also be useful to warn the users that depending on the needs of the user, this may require some more prompt engineering.</em>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>"B1 level of English" may not be universally understood -- it may be better to specific an age (e.g. 5-year-old of 10-year-old)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after #### </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>I'd also split this into 3 parts: before/prompt template/after.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after #### </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header needs a space after ####</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Sentiment Analysis should be lowercase</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    PROJECT_ID = "[your-project-id]"  # @param {type:"string"}
</pre><p>Could you change the line to the way I've written it above?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>good shout thanks, renamed to configure to be more consistent with documentation/blogs</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

[the remark this answers is not in the document]
Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

[the remark this answers is not in the document]
Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>done - thanks</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks - done</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Good shout thank you! I tested this and it failed as expected. I've added the below text. Please shout if you know of another notebook that has a good example of this:</p><p> </p><p><strong>Note:</strong>&nbsp;If you are using Vertex AI workbench, the service account used by Vertex AI may not have sufficient permissions to add IAM policy bindings.</p><p>The&nbsp;<a href="https://colab.research.google.com/corgiredirector?site=https%3A%2F%2Fcloud.google.com%2Fiam%2Fdocs%2Fgranting-changing-revoking-access%23grant-single-role&amp;link_redirector=1" target="_blank">IAM Grant Access</a>&nbsp;page gives instructions on how these policy bindings can be added using Cloud Shell.</p>

[the remark this answers is not in the document]
Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>removed</p>

[the remark this answers is not in the document]
Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>cheers - updated by Alicia</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>cheers - updated by Alicia</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>thanks for this - updated by Alicia</p>

Comment (polong-lin) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Ah, "executing LLMs" sounds a bit strange. I think it might be better to say "Using LLMs with BigQuery ML" and "To use LLMs in BQML, ..."</p>

Comment (polong-lin) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Ah could you rephrase using second-party pronouns as in "In this case, the LLM responded to the simple prompt to describe a cat in one paragraph"</p><p> </p><p>and</p><p> </p><p>"For the rest of the examples, you can ..."</p>

File: language/examples/bqml/bqml_llm_examples.ipynb [hunk 2]
```diff
@@ -0,0 +1,1114 @@
...
```
Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Could the title be something more like:</p><p>Intro to using BigQuery with Vertex AI LLMs</p><p> </p><p>to be a bit more descriptive without over-using acronyms?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <ol><li>Could you add links to the products or product documentation?</li><li>Also Vertex LLMs should perhaps be "Vertex AI LLMs"</li><li>Could you also elaborate a little more about the value of using LLMs on data in BigQuery? We might not be able to expect everyone coming to this repo will be familiar with Google Cloud products and it'd be good to just give a bit more of a preamble to set the context a bit more (or point to relevant docs).</li><li>I'd also remove the emphasis on "NLP" as I think the terminology has shifted towards "generative AI" as in "applying LLMs to text and other structured data in BigQuery"</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Does the REGION need to be a specific region to work, with the remote connection (e.g. does it have to be "US" or "us-central1" for example?? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Might be helpful to add some comments under "try" and "except" to explain what's happening in this block of code?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Capitalize: Vertex AI workbench --&gt; Vertex AI <strong>W</strong>orkbench</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Can the dataset be any region, or should it match the same region as the resource connection that was just created above (and if so, it might be good to explicitly specify it here)?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>I'd probably shy away from using the term "NLP" tasks as it can get confused with the Natural Language API integration with BQML</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Avoid "NLP tasks" which sound like they would not be generative in nature.</p><p> </p><p>Typo: generatave AI -&gt; generative AI</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>good spot. Had another look and this is actually the region for the BQ resource connection, not necessarily Vertex AI directly. Updated with example and link</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>updated - ta</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>updated, ta</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>yes good shout thank you, made this more explicit.</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>updated, ta</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>updated, ta</p>

Comment (polong-lin) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Great -- though typo: datset --&gt; dataset</p>

File: language/examples/bqml/bqml_llm_examples.ipynb [hunk 3]
```diff
@@ -0,0 +1,1131 @@
...
```
Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Dataset ID should be DATASET_ID to keep consistent in styling with the bullet points thereafter perhaps?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

File: language/examples/bqml/bqml_llm_examples.ipynb (hunk 2 above)
Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: updated - ta

File: language/examples/bqml/bqml_llm_examples.ipynb (hunk 3 above)
Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: updated - ta

Comment (polong-lin) [inline language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Header here should also be "Using LLMs with BigQuery ML" perhaps</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/127/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>updated - thanks!</p>

File: language/examples/bqml/bqml_llm_examples.ipynb (hunk 1 above)
Comment (polong-lin) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>Not a must-have, but just wanted to seek any final thoughts on this before resolving this comment -- cheers!</p>

Comment (rachael-ds) [reply language/examples/bqml/bqml_llm_examples.ipynb:1]: <p>heya, Alicia added the before/after and word count difference, we also added this disclaimer after the code block: <em>The word count of the results may not always be within the the 25 words requested and so further&nbsp;</em><a href="https://github.com/GoogleCloudPlatform/generative-ai/blob/main/language/intro_prompt_design.ipynb" target="_blank"><em>prompt engineering</em></a><em>&nbsp;may be required.  </em>Shout if we've missed anything</p><p> </p>

File: language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb [hunk 4]
```diff
@@ -0,0 +1,1087 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "5JzOqTnIizNw"
+   },
+   "source": [
+    "#### Sentiment Analysis\n",
+    "\n",
+    "This example runs sentiment analysis on movie reviews stored in the BigQuery IMDB public dataset to determine whether the movie is positive, negative or neutral."
```
Comment (alokpattani) [inline language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb:826]: determine whether the movie is positive -> determine whether the movie _review_ is positive

File: language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb [hunk 5]
```diff
@@ -0,0 +1,1087 @@
...
+    "id": "4ALWl_i4oF-X"
+   },
+   "outputs": [],
+   "source": [
+    "PROMPT = \"Please categorize this movie review as either Positive, Negative or Neutral. Here is an example. Review: I dislike this movie, Sentiment: Negative\"\n",
+    "\n",
+    "sql = f\"\"\"\n",
+    "          SELECT\n",
+    "            review,\n",
+    "            CONCAT('{PROMPT}',' Review: review Category:') as prompt_template,\n",
```
Comment (alokpattani) [inline language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb:842]: Probably want to change "Category:" to "Sentiment:" in this prompt or change the example from "Sentiment: Negative" to "Category: Negative"

**Review by alokpattani — APPROVED** [review]
This notebook looks solid and runs fine in Colab (based on my testing, anyway). Just added a few minor comments throughout.

File: language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb [hunk 6]
```diff
@@ -0,0 +1,1087 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "n_RiJpQX1BuX"
+   },
+   "source": [
+    "### Objectives\n",
+    "The objective is to demonstrate some of the many ways LLMs can be applied to your BigQuery data using BigQuery ML.\n",
+    "\n",
+    "\n",
+    "You will execute simple SQL statements that call the Vertex AI API with the (`ML.GENERATE_TEXT`) function to:\n",
```
[the remark this answers is not in the document]
Comment (polong-lin) [reply language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb:78]: Ty!

File: language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb (hunk 4 above)
Comment (polong-lin) [reply language/use-cases/applying-llms-to-data/bigquery_ml_llm.ipynb:826]: TY
